### ISSUE TITLE
AWS Profile in kubeconfig template

### ISSUE THREAD

**Issue by mschuchard** [terraform-aws-modules/terraform-aws-eks#39]
## I'm submitting a

* [x] feature request

For my current delivery, the customer has credentials with multiple profiles and not only needs to specify different profiles per cluster, but has no `default` profile.

It would be great if the `kubeconfig.tpl` could be modified:

```yaml
...
users:
- name: aws
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      command: heptio-authenticator-aws
      args:
        - "token"
        - "-i"
        - "${cluster_name}"
      env:
        - name: AWS_PROFILE
          value: ${aws_profile}
```

where the default Terraform value to populate the template would be `default` to ensure no regression.

I wanted to start a discussion before a PR to ensure best path forward on this. Thanks!

**Comment by brandoconnor** [comment]
Hi @mschuchard 

This sounds good but I think this is a duplicate request. We just started discussing a similar (if not the same) issue [here ](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/33)a day ago. Let me know if your use case can be accommodated by implementing what's proposed there.

**Comment by mschuchard** [comment]
I do not see any mention of implementing AWS profiles into the kubeconfig in that issue. Version 1.1.0 is currently failing for the customer because EKS cannot be reached due to the kubeconfig not using the correct AWs profile since that is not customizable in this module.

Having separate kubeconfigs is fine, but in reality people are going to be merging their kubeconfigs and then differentiating between them via `context`.

**Comment by sdavids13** [comment]
I submitted PR #49 which provides the ability to specify a generic map of environment variables to be appended to the kubeconfig.

**Comment by brandoconnor** [comment]
This is addressed having just merged #49 . Closing this up.
